src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -217,23 +221,32 @@ protected Object invokeStep(StepDescriptor d, Object args) {
      * @param args The arguments passed to the step.
      */
     protected Object invokeStep(StepDescriptor d, String name, Object args) {
-        final NamedArgsAndClosure ps = parseArgs(args, d);
-
-        CpsThread thread = CpsThread.current();
-
-        FlowNode an;
 
         // TODO: generalize the notion of Step taking over the FlowNode creation.
         boolean hack = d instanceof ParallelStep.DescriptorImpl || d instanceof LoadStep.DescriptorImpl;
 
-        if (ps.body == null && !hack) {
+        FlowNode an;
+        CpsThread thread = CpsThread.current();
+        boolean hasBody = false;
+        if (args.getClass().isArray()) {
+            for (int i = 0; i < Array.getLength(args); i++) {
+                if (Array.get(args, i) instanceof CpsClosure) {
+                    hasBody = true;
+                    break;
+                }
+            }
+        }
+        if (!hasBody && !hack) {
```

Comment:
Checking `args`  for any `CpsClosure`  to indicate if there is a body. Hopefully this will address the issue in https://github.com/jenkinsci/workflow-cps-plugin/pull/370#discussion_r463835954